xonsh/tools.py

```diff
@@ -941,7 +943,10 @@ def ensure_slice(x):
         else:
             raise ValueError('cannot convert {!r} to slice'.format(x))
     except TypeError:
-        raise TypeError('ensure_slice() argument must be a string or a number not {}'.format(type(x)))
+        try:
+            s = slice(*(int(i) for i in x))
+        except (TypeError, ValueError):
+            raise ValueError('cannot convert {!r} to slice'.format(x))
```

Comment:
i gave a thought on that. i feel like TypeError would be if we didnt accept the type, its the value of x that we try to make into a slice and complain if it cannot be explicitly done. in short, we dont care about the type.